src/main/java/com/google/api/codegen/transformer/py/PythonPackageMetadataTransformer.java

```diff
@@ -0,0 +1,161 @@
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Responsible for producing package metadata related views for Python */
+public class PythonPackageMetadataTransformer implements ModelToViewTransformer {
+
+  PackageMetadataConfig packageConfig;
+
+  public PythonPackageMetadataTransformer(PackageMetadataConfig packageConfig) {
+    this.packageConfig = packageConfig;
+  }
+
+  @Override
+  public List<ViewModel> transform(final Model model, final ApiConfig apiConfig) {
+    String version = packageConfig.apiVersion();
+    List<ViewModel> metadata =
+        computeInitFiles(computePackages(apiConfig.getPackageName()), version);
+    for (String templateFileName : getTopLevelTemplateFileNames()) {
+      metadata.add(generateMetadataView(model, apiConfig, templateFileName));
+    }
+    return metadata;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    List<String> templates = new ArrayList<>();
+    templates.addAll(getTopLevelTemplateFileNames());
+    templates.addAll(getInitTemplateFileNames());
+    return templates;
+  }
+
+  public List<String> getTopLevelTemplateFileNames() {
+    return Lists.newArrayList(
+        "LICENSE.snip",
+        "py/MANIFEST.in.snip",
+        "py/PUBLISHING.rst.snip",
+        "py/setup.py.snip",
+        "py/requirements.txt.snip",
+        "py/README.rst.snip",
+        "py/tox.ini.snip");
+  }
+
+  public List<String> getInitTemplateFileNames() {
+    return Lists.newArrayList("py/__init__.py.snip", "py/namespace__init__.py.snip");
+  }
+
+  private ViewModel generateMetadataView(Model model, ApiConfig apiConfig, String template) {
+    int extensionIndex = template.lastIndexOf(".");
+    String outputPath = template.substring(0, extensionIndex);
+
+    return PackageMetadataView.newBuilder()
+        .templateFileName(template)
+        .outputPath(outputPath)
+        .identifier(apiConfig.getDomainLayerLocation())
+        .packageVersionBound(packageConfig.packageVersionBound(TargetLanguage.PYTHON))
+        .protoPath(packageConfig.protoPath())
+        .shortName(packageConfig.shortName())
+        .gaxVersionBound(packageConfig.gaxVersionBound(TargetLanguage.PYTHON))
+        .protoVersionBound(packageConfig.protoVersionBound(TargetLanguage.PYTHON))
+        .commonProtosVersionBound(packageConfig.commonProtosVersionBound(TargetLanguage.PYTHON))
+        .packageName(packageConfig.packageName(TargetLanguage.PYTHON))
+        .majorVersion(packageConfig.apiVersion())
+        .author(packageConfig.author())
+        .email(packageConfig.email())
+        .homepage(packageConfig.homepage())
+        .licenseName(packageConfig.licenseName())
+        .fullName(model.getServiceConfig().getTitle())
+        .serviceName(apiConfig.getPackageName())
+        .namespacePackages(
+            computeNamespacePackages(apiConfig.getPackageName(), packageConfig.apiVersion()))
+        .hasMultipleServices(false)
+        .build();
```

Comment:
This code is kind of a long stretch to copy for every language...

I don't think you need a metadata namer. You could just pass in TargetLanguage as an argument.

Here are some ideas for strategies for setting language-specific bits:

1. Have the transformer take an empty Builder for PackageMetadataView instead of creating its own fully-constructed instance; the caller can then fill in its own bits here & there and call build() itself.
2. Pass in the extra language-specific things as arguments. (This unfortunately is likely to grow)
3. Create language-specific specialty views: PythonExtraMetadata (holds namespacePackages), NodeExtraMetadata (holds hasMultipleServices, identifier)